apollo-normalized-cache-api/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/api/internal/Normalizer.kt

```diff
@@ -83,14 +89,16 @@ internal class Normalizer(
       } else {
         key
       }
-
-      fieldKey to replaceObjects(
+      val value = replaceObjects(
           entry.value,
           mergedField,
           mergedField.type,
           base.append(fieldKey),
           embeddedFields
       )
+      val arguments = mergedField.argumentsWithValue(variables)
+      val metadata = metadataGenerator.metadataForObject(entry.value, MetadataGeneratorContext(field = mergedField, variables))
```

Comment:
Would it be possible to store `arguments` in metadata? Looks like they're both computed in the same place and `metadataGenerator` has access to the arguments if it wants to.

This way it stores less data in the `Record` and the API is simpler?

Reply:
Oh yeah, the `MetadataGenerator` can store arguments values if it needs them, but we can avoid doing it in the general case. I think that works 👍.